### PR TITLE
My reservations

### LINKED ISSUE #12: [2pt] My reservations page
- [ ] Create a page to show a list of reservations for the user.

### PR DESCRIPTION
**Pull request by Achilles-Dev**
## Changes in this PR:
- [ ] Created the `My Reservations` Page.
- [ ] Merged pull request [fix-redux-issues](https://github.com/AyoubMs/book-an-appointment-front-end/pull/5)
  [x] closes https://github.com/sentayhu19/Book-an-appointment-back-end/issues/12

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,26 @@
...
-        <Navigation />
-      </Router>
+    <div className="App d-flex">
+      <Navigation />
+      <Routes>
+        <Route
+          exact="true"
+          path="/reservations"
+          element={
+          currentUser.token && currentUser.token !== null
```
Comment (AyoubM490) [inline src/App.js:18]: checking for currentUser.token is enough, no need for rechecking with currentUser.token !== null

File: src/components/reservations/MyReservationsPage.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchReservations } from '../../redux/reservations/reservations';
+import Reservation from './Reservation';
+
+const MyReservationsPage = ({ userId }) => {
+  const reservations = useSelector((state) => state.reservations);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
```
Comment (AyoubM490) [inline src/components/reservations/MyReservationsPage.js:11]: good job using useEffect to prevent infinite loop

File: src/index.js [hunk 3]
```diff
@@ -5,6 +5,13 @@ import { Provider } from 'react-redux';
 import { BrowserRouter as Router } from 'react-router-dom';
 import App from './App';
 import store from './redux/configureStore';
+import { login } from './redux/auth';
```
Comment (AyoubM490) [inline src/index.js]: as this is used for test purposes, please remove login

File: src/layout/navigation/Navigation.css [hunk 4]
```diff
@@ -3,12 +3,8 @@
   flex-direction: column;
   justify-content: space-between;
   padding: 20px 15px;
-  height: 100%;
+  height: 100vh;
   width: 300px;
```
Comment (AyoubM490) [inline src/layout/navigation/Navigation.css]: good job using vh, please change the width to a percentage (30% for example) to make the UI responsive

File: src/redux/api.js [hunk 5]
```diff
@@ -45,8 +45,8 @@ const API = {
         success(error.message);
       });
   },
-  addMotor: (motor, success) => {
-    axios.post(`${baseURL}/motorcycles`, {
+  addMotor: (motor, userId, success) => {
```
Comment (AyoubM490) [inline src/redux/api.js:48]: good job fixing the API functions

**Review by AyoubM490 — CHANGES_REQUESTED** [review]

**Comment by AyoubM490** [comment]
Please fix the linter errors

File: src/App.js (hunk 1 above)
Comment (Achilles-Dev) [reply src/App.js:18]: I believe it is necessary because the token is already set to null. So any change from null means a user is present.

**Review by Debas-31 — APPROVED** [review]
## Status Approved :green_circle: 
>## Hello @Achilles-Dev  :wave: 
Your project is complete, and you have implemented the requirement and suggestions the previous reviewer gave. So you can merge and go to the next project.

Happy coding :+1:

**Review by AyoubM490 — APPROVED** [review]
Good job implementing the changes.
You project is complete, and it's time to merge!

**Pull request merged**